src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -21,8 +30,104 @@ public function getUrl(string $bucket, string $key): string
         return $this->getEndpoint($uri, [], null);
     }
 
-    /*
-     * TODO Add get() and put() which supports large files
-     * TODO add has()
+    public function download(string $bucket, string $key): ResultStream
+    {
+        return $this->getObject(['Bucket' => $bucket, 'Key' => $key])->getBody();
+    }
+
+    public function has(string $bucket, string $key): bool
+    {
+        return $this->objectExists(['Bucket' => $bucket, 'Key' => $key])->isSuccess();
+    }
+
+    /**
+     * @param string|resource|callable|iterable $object
+     * @param array{
+     *   ACL?: \AsyncAws\S3\Enum\ObjectCannedACL::*,
+     *   CacheControl?: string,
+     *   ContentType?: string,
+     *   Metadata?: array<string, string>,
```

Comment:
We could add the `content-length` option and use it to get the `$contentLength` when the stream is a callable

Reply:
That would just be a tiny optimisation. 

I also dont expect people that using a callable to know the exact length of the result.

Reply:
Added